the-collab-lab/tcl-23-smart-shopping-list
# Issue #3 | Creates views  "add" & " list", and routes between them

## Description

**What does this code change:**
This adds two different views and routes so that the user is able to switch views (rendering different components) through clicking the corresponding links. Aside from being able to render these different components through their own routes, the styling of the links changes according to the view being rendered. To achieve the styling change, our first approach was to implement a CSS class toggle while tracking state, but after checking out the `react router` documentation we made use of the `<NavLink>` tag and achieved the same result implementing a much cleaner solution. It was a great learning opportunity to try something new and make good use of the documentation.


## Related Issue

Closes #3 

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<img width="657" alt="00_UI_before" src="https://user-images.githubusercontent.com/20430736/113787681-dc3aa080-9711-11eb-95c6-5e41ee9f78d0.png">



### After

<img width="595" alt="01_UI_after" src="https://user-images.githubusercontent.com/20430736/113787708-e52b7200-9711-11eb-97ef-3fd0cbe49e9c.png">
<img width="595" alt="02_UI_after" src="https://user-images.githubusercontent.com/20430736/113787712-e65c9f00-9711-11eb-8985-0e3ee2e94d2f.png">



## Testing Steps / QA Criteria

* Switch from the main branch to the `jc-ar-create-routes` branch
* Pull the changes from this branch
* Run `npm update` to make sure the current dependencies are installed
* Run `npm start` to run the local host
* Go to localhost:3000 in your browser
